routing/router.go

```diff
@@ -345,54 +324,39 @@ func (r *ChannelRouter) syncGraphWithChain() error {
 
 // networkHandler is the primary goroutine for the ChannelRouter. The roles of
 // this goroutine include answering queries related to the state of the
-// network, syncing up newly connected peers, and also periodically
-// broadcasting our latest state to all connected peers.
+// network, pruning the graph on new block notification, applying network
+// updates, and registering new topology clients.
 //
 // NOTE: This MUST be run as a goroutine.
 func (r *ChannelRouter) networkHandler() {
 	defer r.wg.Done()
 
-	var announcementBatch []lnwire.Message
-
-	// TODO(roasbeef): parametrize the above
-	trickleTimer := time.NewTicker(time.Millisecond * 300)
-	defer trickleTimer.Stop()
-
-	retransmitTimer := time.NewTicker(time.Minute * 30)
-	defer retransmitTimer.Stop()
-
 	for {
 		select {
-		// A new fully validated network message has just arrived. As a
+		// A new fully validated network update has just arrived. As a
 		// result we'll modify the channel graph accordingly depending
 		// on the exact type of the message.
-		case netMsg := <-r.networkMsgs:
-			// Process the network announcement to determine if
-			// this is either a new announcement from our PoV or an
-			// update to a prior vertex/edge we previously
+		case updateMsg := <-r.networkUpdates:
+			// Process the routing update to determine if this is
+			// either a new update from our PoV or an update to a
+			// prior vertex/edge we previously
 			// accepted.
-			accepted := r.processNetworkAnnouncement(netMsg.msg)
-
-			// If the update was accepted, then add it to our next
-			// announcement batch to be broadcast once the trickle
-			// timer ticks gain.
-			if accepted {
-				// TODO(roasbeef): exclude peer that sent
-				announcementBatch = append(announcementBatch, netMsg.msg)
-
-				// Send off a new notification for the newly
-				// accepted announcement.
-				topChange := &TopologyChange{}
-				err := addToTopologyChange(r.cfg.Graph, topChange,
-					netMsg.msg)
-				if err != nil {
-					log.Errorf("unable to update topology "+
-						"change notification: %v", err)
-				}
+			err := r.processUpdate(updateMsg.msg)
+			updateMsg.err <- err
+			if err != nil {
+				continue
+			}
 
-				if !topChange.isEmpty() {
-					r.notifyTopologyChange(topChange)
-				}
+			// Send off a new notification for the newly
+			// accepted update.
+			topChange := &TopologyChange{}
+			err = addToTopologyChange(r.cfg.Graph, topChange,
+				updateMsg.msg)
+			if err != nil {
+				log.Errorf("unable to update topology "+
+					"change notification: %v", err)
+			} else if !topChange.isEmpty() {
```

Comment:
If a `continue` is inserted at the line above, then the `topChange.isEmpty()` fragment can be moved out from under this `if/else` block.